tests/ServerTest.php

```diff
@@ -1023,6 +1023,162 @@ function ($data) use (&$buffer) {
         $this->assertInstanceOf('InvalidArgumentException', $error);
     }
 
+    public function testInvalidChunkHeaderResultsInErrorResponse()
```

Comment:
Afaict this does not "result in error reponse", but actually "result in an error on the request stream"?

These tests (see below) LGTM otherwise, but I'm not sure if they provide much value in `ServerTest` or if we should re-organize this? Any input would be welcome :+1: